File: slack-poster/src/main/java/dev/number6/slackposter/dagger/SlackPosterComponent.java
package dev.number6.slackposter.dagger;

import dev.number6.slack.dagger.AWSSecretsModule;
import dev.number6.slack.dagger.HttpModule;
import dev.number6.slackposter.SlackService;
import dagger.Component;

@Component(modules = {SlackPortModule.class,
        SlackPosterConfigurationModule.class,
        HttpModule.class})
public interface SlackPosterComponent {

    SlackService handler();
}

File: slack-poster/src/main/kotlin/dev/number6/slackposter/dagger/SlackPosterComponent.kt
package dev.number6.slackposter.dagger

import dagger.Component
import dev.number6.slack.dagger.SlackClientLiveBindingsModule
import dev.number6.slackposter.SlackService

@Component(modules = [SlackPosterPortModule::class,
    SlackPosterConfigurationModule::class,
    SlackClientLiveBindingsModule::class])
interface SlackPosterComponent {
    fun handler(): SlackService
}